crates/revm/src/evm_impl.rs

```diff
@@ -554,12 +554,23 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> EVMImpl<'a, GSPEC, DB,
                     && bytes.len() > self.data.env.cfg.limit_contract_code_size.unwrap_or(0x6000)
                 {
                     self.data.journaled_state.checkpoint_revert(checkpoint);
-                    return (
-                        InstructionResult::CreateContractSizeLimit,
-                        ret,
-                        interpreter.gas,
-                        bytes,
-                    );
+                    return if INSPECT {
+                        self.inspector.create_end(
+                            &mut self.data,
+                            inputs,
+                            InstructionResult::CreateContractSizeLimit,
+                            ret,
+                            interpreter.gas,
+                            bytes,
+                        )
+                    } else {
+                        (
+                            InstructionResult::CreateContractSizeLimit,
+                            ret,
+                            interpreter.gas,
+                            bytes,
+                        )
+                    };
```

Comment:
This is not exactly the pattern used elsewhere

https://github.com/bluealloy/revm/blob/5d6ecd03ca8cf749b5f0d38a95fa628c32fc8387/crates/revm/src/evm_impl.rs#L405-L412

by default `create_end` would return `::Continue`

Reply:
Isn't Inspector::create_end supposed to be called for every possible reason the create function might end?

Reply:
> by default `create_end` would return `::Continue`

And doesn't the default implementation of `create_end` return `ret`?

https://github.com/bluealloy/revm/blob/5d6ecd03ca8cf749b5f0d38a95fa628c32fc8387/crates/revm/src/inspector.rs#L123